docs/resources/aws_security_group.md.erb

```diff
@@ -134,7 +200,104 @@ A String in the format 'vpc-' followed by 8 hexadecimal characters reflecting VP
+  * protocol - Specifies the IP protocol. 'tcp', 'udp', and 'icmp' are some typical values. The string "-1" or 'any' is used to indicate any protocol.
+  * to_port - Determines if there is a rule whose port range ends at the specified number. The word 'to_' does *not* relate to inbound/outbound directionality; it relates to the port range ("counting _to_"). `to_port` is an exact criterion; so if the rule allows 1000-2000 and you specify a `to_port` of 1999, it will not match.
+  
+    describe aws_security_group(group_name: 'mixed-functionality-group') do
+      # Allow RDP from certain range
+      it { should allow_in(port: 3389, ipv4_range: '10.5.0.0/16')}
+
+      # Allow SSH from two ranges
+      it { should allow_in(port: 22, ipv4_range: ['10.5.0.0/16', '10.2.3.0/24'])}
+
+      # Check Bacula port range
+      it { should allow_in(from_port: 9101, to_port: 9103, ipv4_range: '10.6.7.0/24')}
+
+      # Assuming the AWS SG allows 9001-9003, use inexact matching to check 9002
+      it { should allow_in(port: 9002)}
+
+      # Assuming the AWS SG allows 10.2.1.0/24, use inexact matching to check 10.2.1.33/32
+      it { should allow_in(ipv4_range: '10.2.1.33/32')}
+
+      # Ensure the 3rd outbound rule is TCP-based
+      it { should allow_in(protocol: 'tcp', position: 3')}
+
+      # Don't allow any service to be open to the world
+      it { should_not allow_in(ipv4_range: '0.0.0.0/0')}
+    end
+
+    # Suppose you have a group that should allow SSH and RDP from 
+    # the admin network, 10.5.0.0/16. The resource has 2 rules to 
+    # allow this, and you want to ensure no others have been added.
+    describe aws_security_group(group_name: 'admin-group') do
+      # Allow RDP from certain range and nothing else
+      # The SG must have this rule in position 1 and it must match this exactly
+      it { should allow_in_only(port: 3389, ipv4_range: '10.5.0.0/16', position: 1)}
+
+      # Specify position 2 for the SSH rule.  Without `position`, 
+      # allow_in_only will only allow there to be one rule total.
+      it { should allow_in_only(port: 22, ipv4_range: '10.5.0.0/16', position: 2)}
+
+      # Because this is an _only matcher, this will fail - _only matchers 
+      # use exact IP matching.
+      it { should allow_in_only(port: 3389, ipv4_range: '10.5.1.34/32', position: 1)}
+    end
+    
+### be\_open\_to\_the\_world
+
+This high-level matcher will cause the control to pass if there is any inbound rule that is open to `0.0.0.0/0`, regardless of port.  This is intended to be used with `should_not`.
+
+Note that as of this release, this matcher does not consider:
+  * Worldwide IPv6 access
+  * Transitive access from other unsecured security groups
+  * VPC peering rules to unsecure VPCs
+
+    describe aws_security_group(group_name: 'no-access') do
+      it { should_not be_open_to_the_world }
```

Comment:
Most likely will need to deprecate `be_open_to_the_world` when aliens land we take more of a `universe` perspective. 😜